### PR TITLE
Imported Hatari

### PR DESCRIPTION
**Pull request by jacobzimmermann**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* --device=all to allow using game controllers
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
use_xdg_config_home.patch makes Hatari store its config in .var/app/net.sourceforge.hatari, it would not be meaningful upstream.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.tuxfamily.hatari

File: org.tuxfamily.hatari.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		<p>Hatari supports the emulation of most of the ST and STE hardware, including some special tricks like raster interrupts, border effects and PSG sample sound. A joystick can be emulated with the cursor keys or with a real PC joystick. Hatari also features the possibility to "mount" a directory of the host system as a ST drive, or to use images from an original Atari's hard disk.
+Hatari is also able to emulate the additional hardware of the TT or Falcon machines, like the Videl video chip or the DSP 56001. However, note that TT and Falcon support are currently still in development and experimental.</p>
+	</description>
+
+	<url type="homepage">https://hatari.tuxfamily.org</url>
+	<project_license>GPL</project_license>
```
Comment (hfiguiere) [inline org.tuxfamily.hatari.appdata.xml:15]: This SPDX license identifier is invalid. See https://spdx.org/licenses/

**Comment by hfiguiere** [comment]
bot, build org.tuxfamily.hatari

**Comment by hfiguiere** [comment]
The build log says:
```
  - portaudio :	not found, install it to enable the Falcon microphone input
  - capsimage :	v5 not found, install it to use .IPF, .RAW and .CTR disk images
```

You might want to consider these dependencies.

File: org.tuxfamily.hatari.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+
+command: hatari
+finish-args:
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=ipc
+  - --device=all
```
Comment (hfiguiere) [inline org.tuxfamily.hatari.yml:11]: Any specific reason allow `--device=all`

Comment (jacobzimmermann) [reply org.tuxfamily.hatari.yml:11]: I believe this is is required to allow the use of joysticks and game controllers - correct me if I'm wrong

Comment (hfiguiere) [reply org.tuxfamily.hatari.yml:11]: Make sense.

**Comment by jacobzimmermann** [comment]
> The build log says:
> 
> ```
>   - portaudio :	not found, install it to enable the Falcon microphone input
>   - capsimage :	v5 not found, install it to use .IPF, .RAW and .CTR disk images
> ```
> 
> You might want to consider these dependencies.

Added portaudio.
capsimage is redistribuable but not actual Free Software. To my knowledge the IPF, RAW or CTR disk image formats are rarely used in the Atari ST emulation scene.

File: org.tuxfamily.hatari.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+modules:
+  - name: hatari
+    sources:
+      - type: archive
+        url: http://download.tuxfamily.org/hatari/2.3.1/hatari-2.3.1.tar.bz2
+        sha256: 44a2f62ca995e38d9e0874806956f0b9c3cc84ea89e0169a63849b63cd3b64bd
+      - type: patch
+        path: use_xdg_config_home.patch
+    buildsystem: cmake
```
Comment (barthalion) [inline org.tuxfamily.hatari.yml:29]:
```suggestion
    buildsystem: cmake-ninja
```

File: org.tuxfamily.hatari.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    
+    modules:
+      - name: portaudio
+        sources:
+          - type: git
+            url: https://github.com/PortAudio/portaudio.git
+            tag: v19.7.0
+        buildsystem: cmake
```
Comment (barthalion) [inline org.tuxfamily.hatari.yml:37]:
```suggestion
        buildsystem: cmake-ninja
```

**Comment by barthalion** [comment]
Have you contacted upstream about this submission?

**Comment by jacobzimmermann** [comment]
> Have you contacted upstream about this submission?

Yes and I have their approval.

**Comment by hfiguiere** [comment]
> capsimage is redistribuable but not actual Free Software. To my knowledge the IPF, RAW or CTR disk image formats are rarely used in the Atari ST emulation scene.

Make sense. noted.

**Comment by hfiguiere** [comment]
bot, build org.tuxfamily.hatari

**Comment by barthalion** [comment]
bot, build org.tuxfamily.hatari

**Comment by barthalion** [comment]
Should we invite someone from upstream team?

**Comment by barthalion** [comment]
Hey @huth, would you be interested in co-maintaining Hatari on Flathub? There's no obligation you have to, but we'd like you to have write access just in case.

**Comment by huth** [comment]
@barthalion , thanks, I have to admit that I never used flathub before, but if you like, you can add me - in case I get into this topic later.

Another note: Looking through the history of the discussion here, I think you might want to reconsider using the capsimage libary for this build - depending on your target audience. With capsimage enabled, Hatari is not strictly licensed under the GPL2-or-later anymore, but has additional restrictions, e.g. it is not allowed to use such binaries in commercial products.

**Comment by barthalion** [comment]
I think it's fine, we're essentially non-profit as it is now. I will leave the decision to @jacobzimmermann.

/merge @huth

**Comment by jacobzimmermann** [comment]
> I think it's fine, we're essentially non-profit as it is now. I will leave the decision to @jacobzimmermann.
> 
> /merge @huth

I think it's unlikely that a potential commercial product developer would try to reuse binaries out of this flatpak. Since including capsimage provides additional features for users, we may just as well keep it then.

**Comment by huth** [comment]
> > I think it's fine, we're essentially non-profit as it is now. I will leave the decision to @jacobzimmermann.
> > /merge @huth
> 
> I think it's unlikely that a potential commercial product developer would try to reuse binaries out of this flatpak. Since including capsimage provides additional features for users, we may just as well keep it then.

My point is that you explicitely state GPL-2.0+ in the  org.tuxfamily.hatari.appdata.xml file ... this is not true anymore, the license is then GPL-2.0 with additional restrictions. Not sure whether there is a way to express this with SPDX identifiers, though...

**Pull request closed without merging**